packages/p2p/src/components/modal-manager/modal-manager-context-provider.tsx

```diff
@@ -0,0 +1,180 @@
+            });
+        }
+        setIsModalOpen(true);
+    };
+
+    /**
+     * Hides the current shown modal.
+     * If a previous modal was present, by default the previous modal will be shown in-place of the current closed modal.
+     * This option can be overriden by setting `should_hide_all_modals` to `true` in the `options` argument to close all modals instead.
+     *
+     * @param {Object} options - list of supported settings to tweak how modals should be hidden:
+     * - **should_hide_all_modals**: `false` by default. If set to `true`, previous modal will not be shown and all modals are hidden.
+     * - **should_save_form_history**: `false` by default. If set to `true`, form values in modals that has a form with `ModalForm` component
+     * will be saved when the modal is hidden and restored when modal is shown again.
+     */
+    const hideModal = (options?: THideModalOptions) => {
+        // eslint-disable-next-line no-param-reassign
+        if (!options) options = { should_save_form_history: false, should_hide_all_modals: false };
+
+        const { should_save_form_history, should_hide_all_modals } = options;
+
+        if (should_save_form_history) {
+            general_store.saveFormState();
+        } else {
+            general_store.setSavedFormState(null);
+            general_store.setFormikRef(null);
+        }
+
+        if (isDesktop()) {
+            if (should_hide_all_modals || modal.previous_modal) {
+                setModalState({
+                    active_modal: should_hide_all_modals ? null : modal.previous_modal,
+                    previous_modal: null,
+                });
+                if (should_hide_all_modals) setIsModalOpen(false);
+            } else {
+                setModalState({
+                    active_modal: null,
+                });
+                setIsModalOpen(false);
+            }
+        } else if (modal.stacked_modal && Object.keys(modal.stacked_modal).length !== 0) {
+            setModalState({
+                active_modal: should_hide_all_modals ? null : modal.active_modal,
+                stacked_modal: null,
+            });
+            if (should_hide_all_modals) setIsModalOpen(false);
+        } else {
+            setModalState({
+                active_modal: null,
+            });
+            setIsModalOpen(false);
+        }
+    };
+
+    general_store.hideModal = hideModal;
+    general_store.isCurrentModal = isCurrentModal;
+    general_store.modal = modal.active_modal;
+    general_store.showModal = showModal;
+
+    const state: TModalManagerContext = {
+        hideModal,
+        is_modal_open,
+        isCurrentModal,
+        modal: modal.active_modal,
+        previous_modal: modal.previous_modal,
+        stacked_modal: modal.stacked_modal,
+        modal_props,
+        showModal,
+        useRegisterModalProps,
+    };
+
+    return (
+        <ModalManagerContext.Provider value={props.mock ? props.mock : state}>
```

Comment:
`non-blocking:` would this be better?

```suggestion
        <ModalManagerContext.Provider value={props.mock ?? state}>
```